# This file is part of HiGFXback

# requires
REQUIRES="freetype-build jbig2dec-build jpeg-build make-build openjpeg-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
rm -rf thirdparty
sh ./Configure --disable-static --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
install build/*/mudraw $DESTDIR/dfb/bin
install build/*/mutool $DESTDIR/dfb/bin
install -d $DESTDIR/dfb/include/mupdf
install -m 644 fitz/fitz.h $DESTDIR/dfb/include/mupdf
install -m 644 fitz/memento.h $DESTDIR/dfb/include/mupdf
install -d $DESTDIR/dfb/lib
install build/*/libfitz.so.1 $DESTDIR/dfb/lib
ln -sf libfitz.so.1 $DESTDIR/dfb/lib/libfitz.so
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 debian/mupdf.pc $DESTDIR/dfb/lib/pkgconfig

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/mupdf-build.pc << EOF
Name: mupdf
Version: 1.2
Description: PDF rendering library and document manipulation tools
Requires: $REQUIRES

devel=\\
/dfb/include/mupdf/fitz.h \\
/dfb/include/mupdf/memento.h \\
/dfb/lib/libfitz.so \\
/dfb/lib/pkgconfig/mupdf.pc

exec=\\
/dfb/bin/mudraw \\
/dfb/bin/mutool \\
/dfb/lib/libfitz.so.1
EOF

exit

# patch
--- mupdf-1.2.orig/Configure
+++ mupdf-1.2/Configure
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+prefix=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) prefix=${1#*=};;
+    --disable-static) NOSTATIC=no;;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^prefix/s|=.*|= $prefix|" Makefile
+
+sed -i "/^NOSTATIC/s|=.*|= $NOSTATIC|" Makefile
+
+test $PKG_CONFIG || PKG_CONFIG=pkg-config
+if test $PKG_CONFIG_PATH; then
+  sed -i "/^PKG_CONFIG/s|=.*|= PKG_CONFIG_PATH=$PKG_CONFIG_PATH $PKG_CONFIG|" Makefile
+else
+  sed -i "/^PKG_CONFIG/s|=.*|= $PKG_CONFIG|" Makefile
+fi
+
+sed -i "/^prefix/s|=.*|=$prefix|" debian/mupdf.pc
--- mupdf-1.2.orig/Makefile
+++ mupdf-1.2/Makefile
@@ -9,6 +9,10 @@
 
 default: all
 
+NOSTATIC ?= 
+
+PKG_CONFIG ?= pkg-config
+
 # Do not specify CFLAGS or LIBS on the make invocation line - specify
 # XCFLAGS or XLIBS instead. Make ignores any lines in the makefile that
 # set a variable that was set on the command line.
@@ -37,7 +41,7 @@
 CC_CMD = $(QUIET_CC) $(CC) $(CFLAGS) -o $@ -c $<
 CXX_CMD = $(QUIET_CXX) $(CXX) $(CFLAGS) -o $@ -c $<
 AR_CMD = $(QUIET_AR) $(AR) cr $@ $^
-LINK_CMD = $(QUIET_LINK) $(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
+LINK_CMD = $(QUIET_LINK) $(CC) $(LDFLAGS) -o $@ $^
 LINK_V8_CMD = $(QUIET_LINK) $(CXX) $(LDFLAGS) -o $@ $^ $(LIBS_V8)
 MKDIR_CMD = $(QUIET_MKDIR) mkdir -p $@
 
@@ -55,6 +59,9 @@
 	$(AR_CMD)
 	$(RANLIB_CMD)
 
+$(OUT)/libfitz.so.1 :
+	$(LINK_CMD) -shared -Wl,-soname,libfitz.so.1 $(THIRD_LIBS) $(LIBS)
+
 $(OUT)/% : $(OUT)/%.o
 	$(LINK_CMD)
 
@@ -71,7 +78,7 @@
 $(OUT)/%.o : cbz/%.c $(MUCBZ_HDR) | $(OUT)
 	$(CC_CMD)
 $(OUT)/%.o : apps/%.c fitz/fitz.h pdf/mupdf.h xps/muxps.h cbz/mucbz.h | $(OUT)
-	$(CC_CMD)
+	$(CC_CMD) $(X11_CFLAGS)
 $(OUT)/%.o : scripts/%.c | $(OUT)
 	$(CC_CMD)
 
@@ -79,7 +86,11 @@
 
 # --- Fitz, MuPDF, MuXPS and MuCBZ library ---
 
+ifeq "$(NOSTATIC)" ""
 FITZ_LIB := $(OUT)/libfitz.a
+else
+FITZ_LIB := $(OUT)/libfitz.so.1
+endif
 FITZ_V8_LIB := $(OUT)/libfitzv8.a
 
 FITZ_SRC := $(notdir $(wildcard fitz/*.c draw/*.c))
@@ -162,14 +173,24 @@
 # --- Tools and Apps ---
 
 MUDRAW := $(addprefix $(OUT)/, mudraw)
-$(MUDRAW) : $(FITZ_LIB) $(THIRD_LIBS)
+$(MUDRAW) : $(FITZ_LIB)
+ifeq "$(NOSTATIC)" ""
+$(MUDRAW) : $(THIRD_LIBS) $(LIBS)
+endif
 
 MUTOOL := $(addprefix $(OUT)/, mutool)
-$(MUTOOL) : $(addprefix $(OUT)/, pdfclean.o pdfextract.o pdfinfo.o pdfposter.o pdfshow.o) $(FITZ_LIB) $(THIRD_LIBS)
+$(MUTOOL) : $(addprefix $(OUT)/, pdfclean.o pdfextract.o pdfinfo.o pdfposter.o pdfshow.o)
+$(MUTOOL) : $(FITZ_LIB)
+ifeq "$(NOSTATIC)" ""
+$(MUTOOL) : $(THIRD_LIBS) $(LIBS)
+endif
 
 ifeq "$(NOX11)" ""
 MUVIEW := $(OUT)/mupdf
-$(MUVIEW) : $(FITZ_LIB) $(THIRD_LIBS)
+$(MUVIEW) : $(FITZ_LIB)
+ifeq "$(NOSTATIC)" ""
+$(MUVIEW) : $(THIRD_LIBS) $(LIBS)
+endif
 $(MUVIEW) : $(addprefix $(OUT)/, x11_main.o x11_image.o pdfapp.o)
 	$(LINK_CMD) $(X11_LIBS)
 
@@ -211,8 +232,9 @@
 install: $(FITZ_LIB) $(MUVIEW) $(MUDRAW) $(MUTOOL)
 	install -d $(DESTDIR)$(bindir) $(DESTDIR)$(libdir) $(DESTDIR)$(incdir) $(DESTDIR)$(mandir)/man1
 	install $(FITZ_LIB) $(DESTDIR)$(libdir)
+	if test -n "$(NOSTATIC)"; then ln -sf libfitz.so.1 $(DESTDIR)$(libdir)/libfitz.so; fi
 	install fitz/memento.h fitz/fitz.h pdf/mupdf.h xps/muxps.h cbz/mucbz.h $(DESTDIR)$(incdir)
-	install $(MUVIEW) $(MUDRAW) $(MUBUSY) $(DESTDIR)$(bindir)
+	install $(MUVIEW) $(MUDRAW) $(MUTOOL) $(DESTDIR)$(bindir)
 	install $(wildcard apps/man/*.1) $(DESTDIR)$(mandir)/man1
 
 # --- Clean and Default ---
--- mupdf-1.2.orig/Makerules
+++ mupdf-1.2/Makerules
@@ -5,6 +5,11 @@
 
 CFLAGS += -Wall
 
+ifneq "$(NOSTATIC)" ""
+CFLAGS += -fPIC
+LDFLAGS += -Wl,-rpath,$(prefix)/lib
+endif
+
 ifeq "$(build)" "debug"
 CFLAGS += -pipe -g -DDEBUG
 else ifeq "$(build)" "profile"
@@ -24,19 +29,24 @@
 endif
 
 ifeq "$(OS)" "Linux"
-SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
-SYS_OPENJPEG_INC := `pkg-config --cflags libopenjpeg`
-X11_LIBS := -lX11 -lXext
+SYS_FREETYPE_INC := $(shell $(PKG_CONFIG) --cflags freetype2)
+SYS_OPENJPEG_INC := $(shell $(PKG_CONFIG) --cflags libopenjpeg)
+ifeq "$(shell $(PKG_CONFIG) --exists x11 xext && echo yes)" "yes"
+X11_CFLAGS := $(shell $(PKG_CONFIG) --cflags x11 xext)
+X11_LIBS := $(shell $(PKG_CONFIG) --libs x11 xext)
+else
+NOX11=yes
+endif
 endif
 
 ifeq "$(OS)" "FreeBSD"
-SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
+SYS_FREETYPE_INC := $(shell $(PKG_CONFIG) --cflags freetype2)
 LDFLAGS += -L/usr/local/lib
 X11_LIBS := -lX11 -lXext
 endif
 
 ifeq "$(OS)" "SunOS"
-SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
+SYS_FREETYPE_INC := $(shell $(PKG_CONFIG) --cflags freetype2)
 LDFLAGS += -L/usr/local/lib
 X11_LIBS := -lX11 -lXext
 endif
--- mupdf-1.2.orig/debian/mupdf.pc
+++ mupdf-1.2/debian/mupdf.pc
@@ -1,12 +1,9 @@
-prefix=/usr
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+prefix=/usr/local
+libdir=${prefix}/lib
 includedir=${prefix}/include
 
 Name: mupdf
 Description: Library for rendering PDF documents
-Requires.private: freetype2
-Version: 0.5.0
-Libs: -L${libdir} -lmupdf
-Libs.private: -lopenjpeg -ljbig2dec -ljpeg -lz -lm
+Version: 1.2
+Libs: -L${libdir} -lfitz
 Cflags: -I${includedir}
--- mupdf-1.2.orig/fitz/base_trans.c
+++ mupdf-1.2/fitz/base_trans.c
@@ -25,12 +25,11 @@
 blind_horiz(fz_pixmap *tpix, fz_pixmap *opix, fz_pixmap *npix, int time)
 {
 	unsigned char *t, *o, *n;
-	int blind_height, size, span, position, y;
+	int blind_height, span, position, y;
 
 	if (!tpix || !opix || !npix || tpix->w != opix->w || opix->w != npix->w || tpix->h != opix->h || opix->h != npix->h || tpix->n != opix->n || opix->n != npix->n)
 		return 0;
 	span = tpix->w * tpix->n;
-	size = tpix->h * span;
 	blind_height = (tpix->h+7) / 8;
 	position = blind_height * time / 256;
 	t = tpix->samples;
@@ -50,12 +49,11 @@
 blind_vertical(fz_pixmap *tpix, fz_pixmap *opix, fz_pixmap *npix, int time)
 {
 	unsigned char *t, *o, *n;
-	int blind_width, size, span, position, y;
+	int blind_width, span, position, y;
 
 	if (!tpix || !opix || !npix || tpix->w != opix->w || opix->w != npix->w || tpix->h != opix->h || opix->h != npix->h || tpix->n != opix->n || opix->n != npix->n)
 		return 0;
 	span = tpix->w * tpix->n;
-	size = tpix->h * span;
 	blind_width = (tpix->w+7) / 8;
 	position = blind_width * time / 256;
 	blind_width *= tpix->n;
--- mupdf-1.2.orig/fitz/filt_faxd.c
+++ mupdf-1.2/fitz/filt_faxd.c
@@ -152,22 +152,6 @@
 	{-1,0},{-1,0},{-1,0},{-1,0},{-1,0},{-1,0},{-1,0},{-1,0},{-3,3}
 };
 
-/* Uncompressed decoding table. */
-static const cfd_node cf_uncompressed_decode[] = {
-	{64,12},{5,6},{4,5},{4,5},{3,4},{3,4},{3,4},{3,4},{2,3},{2,3},
-	{2,3},{2,3},{2,3},{2,3},{2,3},{2,3},{1,2},{1,2},{1,2},{1,2},{1,2},
-	{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},
-	{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},
-	{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},
-	{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},{0,1},
-	{-1,0},{-1,0},{8,6},{9,6},{6,5},{6,5},{7,5},{7,5},{4,4},{4,4},
-	{4,4},{4,4},{5,4},{5,4},{5,4},{5,4},{2,3},{2,3},{2,3},{2,3},{2,3},
-	{2,3},{2,3},{2,3},{3,3},{3,3},{3,3},{3,3},{3,3},{3,3},{3,3},{3,3},
-	{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},{0,2},
-	{0,2},{0,2},{0,2},{0,2},{0,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},
-	{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2},{1,2}
-};
-
 /* bit magic */
 
 static inline int getbit(const unsigned char *buf, int x)

# source
http://mupdf.com/downloads/archive/mupdf-1.2-source.zip
